src/lib.rs

```diff
@@ -91,3 +96,12 @@ pub unsafe fn JS_ARGV(_cx: *mut JSContext, vp: *mut JSVal) -> *mut JSVal {
 pub unsafe fn JS_CALLEE(_cx: *mut JSContext, vp: *mut JSVal) -> JSVal {
     *vp
 }
+
+// This is measured properly by the heap measurement implemented in SpiderMonkey.
+impl<T: Copy + GCMethods<T>> HeapSizeOf for Heap<T> {
+    fn heap_size_of_children(&self) -> usize {
+        0
+    }
+}
+known_heap_size!(0, JSVal);
```

Comment:
The reason I didn't is that:

* UnsafeCell does not implement HeapSizeOf (and probably shouldn't, if we want to maintain aliasing guarantees). This means I can't derive HeapSizeOf Heap<T>.
* Why bring in a plugin for a one-line derive when a one-line macro will do? This means deriving JSVal is pointless.